core/valor_core/classification.py

```diff
@@ -0,0 +1,1291 @@
+
+    sample_df = pd.concat(
+        [
+            pr_calc_df[pr_calc_df[flag_column]]
+            .groupby(
+                [
+                    "label_key",
+                    "label_value_gt",
+                    "confidence_threshold",
+                ],
+                as_index=False,
+            )[["datum_uid"]]
+            .agg(lambda x: tuple(x.head(max_examples)))
+            .rename(columns={"label_value_gt": "label_value"}),
+            pr_calc_df[pr_calc_df[flag_column]]
+            .groupby(
+                [
+                    "label_key",
+                    "label_value_pd",
+                    "confidence_threshold",
+                ],
+                as_index=False,
+            )[["datum_uid"]]
+            .agg(lambda x: tuple(x.head(max_examples)))
+            .rename(columns={"label_value_pd": "label_value"}),
+        ],
+        axis=0,
+    ).drop_duplicates()
+
+    if not sample_df.empty:
+        sample_df[f"{flag_column}_samples"] = sample_df.apply(
+            lambda row: set(zip(*row[["datum_uid"]])),  # type: ignore - pandas typing error
+            axis=1,
+        )
+
+        pr_curve_counts_df = pr_curve_counts_df.merge(
+            sample_df[
+                [
+                    "label_key",
+                    "label_value",
+                    "confidence_threshold",
+                    f"{flag_column}_samples",
+                ]
+            ],
+            on=["label_key", "label_value", "confidence_threshold"],
+            how="outer",
+        )
+        pr_curve_counts_df[f"{flag_column}_samples"] = pr_curve_counts_df[
+            f"{flag_column}_samples"
+        ].apply(lambda x: list(x) if isinstance(x, set) else list())
+
+    else:
+        pr_curve_counts_df[f"{flag_column}_samples"] = [
+            list() for _ in range(len(pr_curve_counts_df))
+        ]
+
+    return pr_curve_counts_df
+
+
+def _calculate_pr_curves(
+    prediction_df: pd.DataFrame,
+    groundtruth_df: pd.DataFrame,
+    metrics_to_return: list,
+    pr_curve_max_examples: int,
+) -> list[metrics.PrecisionRecallCurve]:
+    """Calculate PrecisionRecallCurve metrics."""
+    if not (
+        enums.MetricType.PrecisionRecallCurve in metrics_to_return
+        or enums.MetricType.DetailedPrecisionRecallCurve in metrics_to_return
+    ):
+        return []
+
+    joint_df = (
+        pd.merge(
+            groundtruth_df,
+            prediction_df,
+            on=["datum_id", "datum_uid", "label_key"],
+            how="outer",
+            suffixes=("_gt", "_pd"),
+        )
+        .assign(
+            is_label_match=lambda chain_df: (
+                (chain_df["label_value_pd"] == chain_df["label_value_gt"])
+            )
+        )
+        .drop(
+            columns=[
+                "annotation_id_gt",
+                "created_at_gt",
+                "annotation_id_pd",
+                "created_at_pd",
+                "label_pd",
```

Comment:
Probably can expand this list of cols to ignore, or just specify the columns you want to _keep_ might be even better

Reply:
all the other columns are used in some way; I've already tried to remove others

Reply:
When I look at the dataframe here I see columns like `bounding_box_gt` and `polygon_gt`, surely those aren't needed in this function?

Reply:
you're right: I thought you were talking about the columns that get dropped for detection tasks. just made this change; thanks for the suggestion